hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/enrichers/UserAgentSpanEnricher.java

```diff
@@ -59,4 +54,14 @@ public void enrichEvent(StructuredTrace trace, Event event) {
           AttributeValueCreator.create(userAgent.getVersionNumber().toVersionString()));
     }
   }
+
+  private Optional<String> getUserAgent(Event event) {
+    if (event.getHttp() != null
+        && event.getHttp().getRequest() != null
+        && !StringUtils.isEmpty(event.getHttp().getRequest().getUserAgent())) {
```

Comment:
This is confusing. There is a getUserAgent in getRequest().getHeaders() too. What is that then?

Reply:
Reading from both `getRequest().getHeaders().getUserAgent()` and `getRequest().getUserAgent()`, preferring the headers user agent